### PR TITLE
Put back pre-commit and add isort

### PR DESCRIPTION
**Pull request by dpoirier**
## Description
Enable pre-commit (you'll still have to run "pre-commit install" in your working directory), and add isort to keep imports nicely sorted.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        args: ["--check"]
+
+  - repo: local
+    hooks:
+      - id: isort
+        name: "Isort"
+        entry: isort --check --diff
+        language: system
+        pass_filenames: true
+        types: ["python"]
```
Comment (Audiosutras) [inline .pre-commit-config.yaml:26]: Can you add a `space` between hooks?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      - id: check-added-large-files
+      - id: check-json
+
+  - repo: https://github.com/psf/black
+    rev: 19.10b0
+    hooks:
+      - id: black
+        args: ["--check"]
+
+  - repo: local
```
Comment (Audiosutras) [inline .pre-commit-config.yaml:19]: I think we should add a local `flake8` hook.

File: Makefile [hunk 3]
```diff
@@ -2,18 +2,16 @@
...
 test:
 	# Run all tests and report coverage
-	# Requires coverage
-	python manage.py makemigrations --dry-run | grep 'No changes detected' || \
-		(echo 'There are changes which require migrations.' && exit 1)
-	coverage run manage.py test --noinput -v 2 --keepdb
-	coverage report -m --fail-under 98
+	pre-commit run -v -a
 
 lint-py:
```
Comment (Audiosutras) [inline Makefile:13]: I think we should remove `lint-py` and add `flake8` as a local hook to `pre-commit`.

File: Makefile [hunk 4]
```diff
@@ -2,18 +2,16 @@
 PROJECT_NAME = cc_licenses
 STATIC_DIR = ./$(PROJECT_NAME)/static
 
-default: lint test
+default:
+	# Checks syntax and runs tests
+	pre-commit run -v -a
 
 test:
```
Comment (Audiosutras) [inline Makefile:9]: Do we need both `default` and `test` if they both run the same command now? Could we just use `default` and point travis to use it?

**Comment by dpoirier** [comment]
@Audiosutras Good comments, I've implemented them.

**Review by Audiosutras — COMMENTED** [review]
Lastly, we want `isort` and `black` to run with the `--check` flag? On each commit or when we run `pre-commit run black`/`pre-commit run isort` we should have the files reformatted instead of just checking that `black` and `isort` would reformat the files.

**Comment by dpoirier** [comment]
Sure, we can have isort and black fix their problems for us.  I've changed that.

I think passing filenames is good because 'pre-commit' will by default only pass the files that are changing in the commit about to be made, which will speed things up.  (We add "-a" in CI to check all the files every time, overriding that, but day-to-day I think I'd rather have my individual commits not too slow unless that's causing a problem.)

**Comment by Audiosutras** [comment]
> Sure, we can have isort and black fix their problems for us. I've changed that.
> 
> I think passing filenames is good because 'pre-commit' will by default only pass the files that are changing in the commit about to be made, which will speed things up. (We add "-a" in CI to check all the files every time, overriding that, but day-to-day I think I'd rather have my individual commits not too slow unless that's causing a problem.)

Sounds good!

**Review by Audiosutras — APPROVED** [review]
💯

**Pull request merged**